### PR TITLE
Add a PR template for github

### PR DESCRIPTION
**Pull request by nyalldawson**
This PR adds a template with a checklist to help guide new QGIS contributors to create quality pull requests. I believe adding this will help new contributors know exactly what's required to get a PR quickly reviewed (and accepted) by QGIS maintainers.

Here's what it looks like:

## Description
Include a few sentences describing the overall goals for this pull request.

## Checklist

All checklist items should be addressed before a QGIS maintainer will review the request:

- [ ] I've read the [QGIS Coding Standards](https://www.qgis.org/en/site/getinvolved/development/qgisdevelopersguide/codingstandards.html) and this PR complies with them
- [ ] This PR passes all existing unit tests
- [ ] New unit tests have been added for core changes
- [ ] I've run the scripts/prepare-commit.sh script before each commit
- [ ] Commit messages are descriptive and explain the rationale for changes
- [ ] Commits which fix bugs include "fixes #11111" in the commit message
- [ ] Commits which add new features are tagged with \[FEATURE\] in the commit message
- [ ] Commits which change the UI or existing user workflows are tagged with \[needs-docs\] in the commit message

### CODE REVIEW

**Comment by DelazJ** [comment]
For your information, there's [a discussion ](https://github.com/qgis/QGIS-Documentation/issues/1566)to move the QGIS Coding Standards to a real and separate developer's guide in doc repository, which led to those two pending PRs (qgis/QGIS-Documentation#1585 and qgis/QGIS-Website#406 )

**Review by sbrunner — APPROVED** [review]

**Review by rldhont — APPROVED** [review]

**Comment by m-kuhn** [comment]
:+1: I really like this

> Commits which fix bugs include "fixes #11111" in the commit message

What about explicitly adding "in addition to the rationale". I hate seeing commit messages that ONLY consist of "fixes #11111"

**Comment by dakcarto** [comment]
Hi @nyalldawson,

Regarding **[ ] This PR passes all existing unit tests**

Since the PR will trigger a run of unit tests via CI and merging is contingent upon a reviewer's approval, isn't this redundant?

Or, are you asking the contributor to verify unit tests pass _before_ submitting the PR? If this is the case, then a potential dev may spend excessive amounts of time trying to get a "fails on their platform" test to pass, whereas it may already pass on CI. Admittedly, it would be best to have all failing tests addressed, but is this necessary prior to a contribution if the test is failing for some other reason than the actual contribution? I don't see any issue with using CI as a unit-test-runner crutch for fledgling contributors.

Essentially, having all unit tests passing locally certainly shouldn't block a contributor from submitting the PR and triggering the project's standardized CI.

Maybe **[ ] This PR passes all relevant unit tests** ? May be hard for a new contributor to know which are relevant, though.

**Comment by m-kuhn** [comment]
Maybe we need to split it into two sections

Before opening a pull request
=======================

[ ] I have read the contributors agreement ...
[ ] etc

Before a review is started
===============

[ ] All unit tests need to pass
[ ] ...

**Comment by DelazJ** [comment]
> All checklist items should be addressed before a QGIS maintainer will review the request

Question: Most of my PR are simple tasks, not involving units tests, no new feature as well... It's more about cosmetics and consistency. And they do not meet half of the checkboxes. So what am I supposed to do? how/when would a reviewer decide it's worth looking? Is there a `not concerned` item available?

**Comment by m-kuhn** [comment]
@DelazJ I don't think the checkboxes will need to be checked. It's more about raising awareness. Let's take as an example your pull request #3953 from this morning: It has a descriptive text and doesn't break any existing tests (e.g. indentation), so I would review it again without any delay.

**Comment by DelazJ** [comment]
> It's more about raising awareness

ok. Thanks

**Comment by m-kuhn** [comment]
I updated the comment [above](https://github.com/qgis/QGIS/pull/3919#issuecomment-270522240).

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+## Description
+Include a few sentences describing the overall goals for this pull request.
```
Comment (m-kuhn) [inline .github/PULL_REQUEST_TEMPLATE.md]: ... and screenshots (if applicable).

**Comment by m-kuhn** [comment]
pull request updated and merged.

I think everyone agrees that it's a good thing to do and we can simply improve on the current version.

@dakcarto I think your points are very valuable, could you add them to the documentation and ping others to fill in the reasoning behind where you are unsure?

**Comment by nyalldawson** [comment]
@m-kuhn thanks! One less open PR is a good thing!

**Pull request closed without merging**